crates/democracy/src/lib.rs

```diff
@@ -490,10 +498,12 @@ pub mod pallet {
             let who = ensure_signed(origin)?;
 
             let seconds = Self::len_of_deposit_of(proposal).ok_or_else(|| Error::<T>::ProposalMissing)?;
+            ensure!(seconds < T::MaxDeposits::get(), Error::<T>::TooMany);
             ensure!(seconds <= seconds_upper_bound, Error::<T>::WrongUpperBound);
             let mut deposit = Self::deposit_of(proposal).ok_or(Error::<T>::ProposalMissing)?;
             T::Currency::reserve(&who, deposit.1)?;
-            deposit.0.push(who);
+            let ok = deposit.0.try_push(who.clone()).is_ok();
+            debug_assert!(ok, "`seconds` is below static limit; `try_insert` should succeed; qed");
```

Comment:
Not a fan of this tbh since in production this would hide unexpected errors. Let's just return an error